nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/xml/XMLReader.java

```diff
@@ -106,12 +107,27 @@ public class XMLReader extends SchemaRegistryService implements RecordReaderFact
             .required(false)
             .build();
 
+    public static final PropertyDescriptor PARSE_XML_ATTRIBUTES = new PropertyDescriptor.Builder()
+            .name("parse_xml_attributes")
+            .displayName("Parse XML Attributes")
+            .description("When 'Schema Access Strategy' is 'Infer Schema' and this property is 'true' then XML attributes are parsed and " +
+                    "added to the record as new fields. When the schema is inferred but this property is 'false', " +
+                    "XML attributes and their values are ignored.")
+            .expressionLanguageSupported(ExpressionLanguageScope.NONE)
+            .allowableValues("true", "false")
+            .defaultValue("true")
+            .required(false)
+            .dependsOn(SchemaAccessUtils.SCHEMA_ACCESS_STRATEGY, INFER_SCHEMA)
+            .build();
+
+    private volatile boolean parseXMLAttributes;
     private volatile String dateFormat;
     private volatile String timeFormat;
     private volatile String timestampFormat;
 
     @OnEnabled
     public void onEnabled(final ConfigurationContext context) {
+        this.parseXMLAttributes = context.getProperty(PARSE_XML_ATTRIBUTES).asBoolean();

```

Comment:
I think camel case may be used in variable naming, even if the class names don't follow the standard.

Reply:
Fixing in my next commit. Thank you!